monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/BufferWithSelectorObservable.scala

```diff
@@ -53,7 +54,8 @@ private[reactive] final class BufferWithSelectorObservable[+A,S](
           upstreamSubscriber.synchronized {
             if (downstreamIsDone) Stop else {
               buffer += elem
-              if (maxSize > 0 && buffer.length >= maxSize)
+              if (maxSize > 0 &&
+                  buffer.foldLeft(0)(_ + sizeOf(_)) >= maxSize) // sum size of elements in the buffer
```

Comment:
I wonder about performance penalty for bigger buffers (goes from O(1) to O(N)). Maybe we should track `buffer` size? What do you think @alexandru - is it something to worry about?

Reply:
@Avasil I agree. The point of having the sizeOf paremeter and in our case was to decrease dramatically the count of elements in the buffer we reach at most 10 elements each containing x100K values so the penalty is very low. But since sizeOf can be hard to compute it certainly worths to care about that.

Reply:
@Avasil Yes I agree with you. Furthermore, sizeOf could be resource consuming so it will surely becomes unacceptable anyway.